loop.py

```diff
@@ -112,8 +152,12 @@ def run_model(r, w, k, theta, D=20, iterations=350):
             print(f"{iteration} Ranking with a total cost of {m.objective_value}")
             print("Number of solutions: %d" % m.num_solutions)
 
-            # array when the value on the i_th position indicates the subject at rank i
+            # array with the value on the i_th position indicates the subject at rank i
             new_ranking = convert_solution_to_ranking(x)
+
+            if k > 1:
+                new_ranking = perform_probabilistic_unfairness_swap(max_unfairness_index, new_ranking, k)
```

Comment:
I assume you know this but you swap LP optimized ranking -> Maybe test it on the normal ranking (without LP)